pkg/external_repos/introspect.go

```diff
@@ -73,6 +74,10 @@ func Introspect(repo *dao.Repository, repoDao dao.RepositoryDao, rpm dao.RpmDao)
 	)
 	log.Debug().Msg("Introspecting " + repo.URL)
 
+	if repo.FailedIntrospectionsCount > FailedIntrospectionsLimit {
+		return 0, fmt.Errorf("introspection skipped because this repository has failed more than %v times in a row", FailedIntrospectionsLimit)
+	}
```

Comment:
we probably only want to do this if its not a 'public' repository, i think we'd want those to keep trying